tuf/repository_tool.py

```diff
@@ -2626,12 +2586,24 @@ def add_target_to_bin(self, target_filepath):
     # types, and that all dict keys are properly named.
     # Raise 'securesystemslib.exceptions.FormatError' if there is a mismatch.
     securesystemslib.formats.PATH_SCHEMA.check_match(target_filepath)
+    tuf.formats.NUMBINS_SCHEMA.check_match(number_of_bins)
+
+    # TODO: check target_filepath is sane
+
+    path_hash = _get_hash(target_filepath)
+    bin_name = _find_bin_for_hash(path_hash, number_of_bins)
+
+    # Ensure the Targets object has delegated to hashed bins
+    if not self._delegated_roles.get(bin_name, None):
+      raise securesystemslib.exceptions.Error(self.rolename + ' does not have'
+          ' a delegated role ' + bin_name)
 
-    return self._locate_and_update_target_in_bin(target_filepath, 'add_target')
+    self._delegated_roles[bin_name].add_target(target_filepath, fileinfo)
```

Comment:
```suggestion
    self._delegated_roles[bin_name].add_target(target_filepath, fileinfo=fileinfo)
```
... otherwise `fileinfo` is passed as `custom` to `add_target`.

It would be great to have a test that for `add_target_to_bin` that adds a `fileinfo`.

Reply:
Great catch Lukas. I've squashed this fix and added a test in 62e4364